src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleMetadata.cs

```diff
@@ -323,44 +323,17 @@ internal static MetadataPropertyName GetMetadataPropertyName(ReadOnlySpan<byte>
             {
                 switch (propertyName.Length)
```

Comment:
I guess this wouldn't be needed if C# `switch` supported utf8 literals and the same trie-like optimization it has for switch-over-string literals 😐